File: Escola/Cursos.Repository/Services/ListaatividadesService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  ListaatividadesService
      /// </summary>
      public partial class ListaatividadesService : BaseServices, IListaatividadesService
      {

		public ListaatividadesService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity ES_LISTAATIVIDADES
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<ES_LISTAATIVIDADES> entity)
        {
            UnitOfWork.ListaatividadesRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity ES_LISTAATIVIDADES
        /// </summary>
        /// <param name="entity"></param>
		public void Add(ES_LISTAATIVIDADES entity)
        {	
            UnitOfWork.ListaatividadesRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity ES_LISTAATIVIDADES
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(ES_LISTAATIVIDADES entity)
        {
            UnitOfWork.ListaatividadesRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity ES_LISTAATIVIDADES
        /// </summary>
        /// <returns>IEnumerable ES_LISTAATIVIDADES</returns>
        public IEnumerable<ES_LISTAATIVIDADES> GetAll()
        {
			return UnitOfWork.ListaatividadesRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity ES_LISTAATIVIDADES
        /// </summary>
        /// <param name="entity"></param>
        public void Update(ES_LISTAATIVIDADES entity)
        {
            UnitOfWork.ListaatividadesRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity ES_LISTAATIVIDADES By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>ES_LISTAATIVIDADES</returns>
		public ES_LISTAATIVIDADES GetById(int id)
        {
			return UnitOfWork.ListaatividadesRepository.FindBy(x => x.ID.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<ES_LISTAATIVIDADES> FindBy(Expression<Func<ES_LISTAATIVIDADES, bool>> predicate)
        {
            return UnitOfWork.ListaatividadesRepository.FindBy(predicate);
        }
      }
}

File: Escola/Cursos.Api/scripts/service/listaatividadesService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/13/2017 1:58:38 PM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("ListaatividadesService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Listaatividades/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Listaatividades/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Listaatividades, callback) {
            $http.post(serviceBase + "Listaatividades/post", Listaatividades)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Listaatividades, callback) {
            $http.post(serviceBase + "Listaatividades/put", Listaatividades)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Listaatividades, callback) {
            $http.post(serviceBase + "Listaatividades/delete", Listaatividades)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

